docs/DeveloperGuide.md

```diff
@@ -352,6 +353,33 @@ Each book request is designed to bind to an isbn instead of a book copy. For exa
 
 Associating a book request with multiple book copies introduces some problems. Initially, when a book becomes available, the same reminder message to the user will be printed multiple times, once per book request per book copy. Nevertheless, this is solved by using a `Set` to store reminder messages so that identical reminder messages will not be added multiple times.
 
+### Book Find feature
+
+This feature allows users to search for any books with either the isbn, title or author's name. 
+
+#### Implementation
+The Book find feature is facilitated by the `BookCommandParser`, `FindBookParser` and `FindBookCommand`. 
+
+Given below is an example usage scenario and how the request mechanism behaves at each step:
+
+1. The user enters a book find command and provides the parameter for the search query.
+2. `LibTaskParser` creates a new `BookCommandParser` after preliminary processing of user input, which in turns creates a new `FindBookParser`.
+3. `FindBookParser` creates either a `BookAuthorContainsKeywordsPredicatem` or `BookNameContainsKeywordsPredicate` or `BookTagContainsKeywordsPredicate` object `predicate` with the search query.4. 
+4. `FindBookParser` creates a new `FindBookCommand` based on the processed input and passes the predicate on.
+5. `LogicManager` then executes the `FindBookCommand`. 
+6. `FindBookCommand` calls `Model#updateFilteredBookList()` with the `predicate`, resulting in the book list to be updated to display all the books that match the given search query.
+7. Finally, the `FindBookCommand` creates a `CommandResult` and returns it to `LogicManager` to complete the command. 
+
+The following activity diagram summarizes what happens when a user executes a request command:
+
+<img src="images/FindBookCommandActivityDiagram.png" width="650" />
+
+
+#### Design considerations
+
+Each find query can only be one of either the tag, author or title. The feature is designed to return all books that match the predicate created and display them. We chose not to include the isbn as a user searchable query as it is likely that most users would remember the title of the book or the author's name rather than the isbn.
```

Comment:
`The feature is designed to return all books` -> can be phrased to -> `The feature is designed to display all books`